### PR TITLE
ci: Build with Java 11 only

### PR DESCRIPTION
**Pull request by NotMyFault**
Can someone replay the build on ci.j please? Otherwise the Jenkinsfile revision is ignored.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -124,6 +124,57 @@
         </plugins>
       </build>
     </profile>
+    <profile>
```
Comment (timja) [inline pom.xml:127]: do we need this profile? this should only be needed for the jenkins parent pom not plugin parent afaik

Comment (jetersen) [reply pom.xml:127]: I don't think we need the profile?

The plugin pom and HPI should do the correct thing?

Comment (NotMyFault) [reply pom.xml]: May be unneeded, but based on my experience, I had to add it when updating the maven stapler plugin to not produce Java 11+ binaries only. Let's see 🤞🏻

Comment (jetersen) [reply pom.xml]: I am fine with breaking java 8 compatibility, would mean we could do a lot of code cleanup! 🗡️

Comment (jetersen) [reply pom.xml]: Would be nice to have JRE metric for plugins:
Much like [Plugin Versions by Jenkins Version](https://stats.jenkins.io/pluginversions)

Comment (timja) [reply pom.xml]: stapler plugin would need it but an actual plugin does not need this

Comment (NotMyFault) [reply pom.xml]: > stapler plugin would need it but an actual plugin does not need this



Removed it, only change that's left is the Jenkinsfile and GH action modification.

**Review by timja — APPROVED** [review]

**Pull request merged**